iib/workers/tasks/iib_static_types.py

```diff
@@ -0,0 +1,136 @@
+from typing import Dict, List, Optional, Set, TypedDict, Union
+
+from operator_manifest.operator import ImageName, OperatorCSV
+
+
+class BundleMetadata(TypedDict):
+    """Type class referencing data of bundle used for mypy checking."""
+
+    found_pullspecs: Set[ImageName]
+    operator_csvs: List[OperatorCSV]
+
+
+class IndexImageInfo(TypedDict):
+    """Type class referencing data related to index image used for mypy checking."""
+
+    arches: Set[str]
+    ocp_version: str
+    resolved_distribution_scope: str
+    resolved_from_index: Optional[str]
+
+
+class AllIndexImagesInfo(TypedDict):
+    """Type class referencing group of IndexImageInfo classes used for mypy checking."""
+
+    from_index: IndexImageInfo
+    source_from_index: IndexImageInfo
+    target_index: IndexImageInfo
+
+
+# MYPY check - should label be optional? If not total parameter can be removed.
+class PrebuildInfo(TypedDict, total=False):
```

Comment:
AFAIK `labels` are indeed optional so it's fine to leave `total=False` there.

Reply:
Do we need those labels? It seems that they are passed to variable `prebuild_info` in the `handle_create_empty_index_request` method, however after that in the `_update_index_image_build_state` method they are not even used {I did not found any other place where those labels could be used}.

Reply:
The element `labels` is a supported optional parameter used only in `create-empty-index` requests, as you can see [here](https://iib.qe.engineering.redhat.com/#/paths/~1builds~1create-empty-index/post). It allows external entities to add additional labels to the resulting index image and if not specified we just set [these ones](https://github.com/release-engineering/iib/blob/master/iib/workers/tasks/build_create_empty_index.py#L126-L129).

If we remove the `labels` from IIB's code it will break this feature from its API on `create-empty-index` and will cause some troubles thus we need to keep it but consider this parameter to be optional from the requester point of view.